### PR TITLE
[ANCHOR-371] Refactor the metrics configuration and the comments of the configuration file.

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

- Remove `network_pass_phrase`
- Refactor the `metrics` section
- Add `management_server_port` to `platform_server`, `event_processor`, and `payment_observer`
- Review and clean up the configuration comments.

### CODE REVIEW

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 1]
```diff
@@ -374,101 +369,91 @@ sep24:
...
   payment_type: STRICT_SEND
 
-  ## @param: deposit_info_generator_type
-  ## @default: self
-  ## Used to choose how the SEP-31 deposit information will be generated, which includes the
-  ##     deposit address, memo and memo type.
-  ## @supported_values:
+  # Used to choose how the SEP-31 deposit information will be generated, which includes the
+  #     deposit address, memo and memo type.
+  # Tje `deposit_info_generator_type` must be one of the following:
```
Comment (philipliu) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:408]: Type: Tje

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 2]
```diff
@@ -374,101 +369,91 @@ sep24:
...
-  ## system)
-  #
-  extras_enabled: false
-  ## @param: run_interval
-  ## @type: integer
-  ## Interval in seconds to query the database to generate metrics
-  #
-  run_interval: 30
+  # The configuration of making extended metrics available.
+  extended_metrics:
```
Comment (philipliu) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:433]: Question: What are extended metrics?

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 3]
```diff
@@ -7,116 +7,133 @@
...
   # The context path of the Stellar observer server
   context_path: /
   # The listening port of the Stellar observer server
   port: 8083
-  ## @param: type
-  ## @type: string
-  ## @supported_values:
-  ##    `stellar`: the observer will observer transactions from the Stellar network
+  # The management server port
+  management_server_port: 9083
```
Comment (philipliu) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:126]: The management server is the metrics server right? Can it be renamed?

Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:126]: The management server is the Java management server exposed by Spring actuator.

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 2 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:433]: They are defined in https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/platform/src/main/java/org/stellar/anchor/platform/service/MetricEmitterService.java#L100-L107

Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:433]: These fields are subject to review and refactor.

**Review by philipliu — APPROVED** [review]
LTGM

File: platform/src/main/resources/config/anchor-config-schema-v1.yaml [hunk 4]
```diff
@@ -95,5 +98,4 @@ sep_server.context_path:
 sep_server.management_server_port:
 sep_server.port:
 stellar_network.horizon_url:
-stellar_network.network:
-stellar_network.network_passphrase:
```
Comment (Ifropc) [inline platform/src/main/resources/config/anchor-config-schema-v1.yaml:99]: Is it something we want to remove in minor version?

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 3 above)
Comment (Ifropc) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:126]: Does it have swagger or something like that? How would developer know how to use management server?
Also, how is it protected by default?

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 5]
```diff
@@ -7,116 +7,133 @@
...
 ##############################
-## Application Configuration
+# Application Configuration
 ##############################
 
+#
+# The default language of the application
+# The supported languages defined by RFC4646 (https://datatracker.ietf.org/doc/html/rfc4646)
+#
+languages: en
```
Comment (Ifropc) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:17]: What does this config do?

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 6]
```diff
@@ -374,101 +369,91 @@ sep24:
...
-## The metrics are exposed at the port specified by sep_server.management_server_port
+# The metrics are exposed at the port specified by each server's (eg: sep, platform, event processor and observer)
+# management_server_port
 metrics:
-  ## @param: enabled
-  ## @type: bool
-  ## If true, enable metrics endpoints for third-party services (ex: prometheus) to scrape
-  #
+  # If true, enable metrics will be enabled.
   enabled: false
```
Comment (Ifropc) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:431]: If it's not enabled, there is nothing running on `management_server_port` right?

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 3 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:126]: The management port is the port used by Spring actuator. 
https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html

It is used for Prometheus to scrape the metrics.

File: platform/src/main/resources/config/anchor-config-schema-v1.yaml (hunk 4 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-schema-v1.yaml:99]: It does not break backward-compatibility. When specified, it is just ignored.

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 3 above)
Comment (Ifropc) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:126]: Can you add it in the code comments please?

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 5 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:17]: The port `9083` is currently used for Spring Boot actuator. The anchor should refer to the Spring documentation.

Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:17]: The `language: en` is used to specify the languages supported in SEP server. It is legacy. Maybe we should review it in the next major release.

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 6 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:431]: According to what we have today [here](https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/platform/src/main/java/org/stellar/anchor/platform/configurator/SepConfigManager.java#L44), that's correct.

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 3 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:126]: This PR is blocking the implementation. I will add the comments with the impl PR.

Comment (Ifropc) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:126]: Sure thank you

**Pull request merged**